modules/stanford_icon_grid/stanford_icon_grid.module

```diff
@@ -0,0 +1,185 @@
+  return FALSE;
+}
+
+/**
+ * Implements hook_field_info().
+ */
+function stanford_icon_grid_field_info() {
+  return array(
+    'fontawesome_icon' => array(
+      'label' => t('Font Awesome Icon'),
+      'description' => t('This field stores varchar text in the database.'),
+      'default_widget' => 'fontawesome_icon_widget',
+      'default_formatter' => 'fontawesome_icon_formatter',
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_widget_info().
+ */
+function stanford_icon_grid_field_widget_info() {
+  return array(
+    'fontawesome_icon_widget' => array(
+      'label' => t('Font Awesome Icon'),
+      'field types' => array('fontawesome_icon'),
+      'behaviors' => array(
+        'multiple values' => FIELD_BEHAVIOR_DEFAULT,
+        'default value' => FIELD_BEHAVIOR_DEFAULT,
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_widget_form().
+ */
+function stanford_icon_grid_field_widget_form(&$form, &$form_state, $field, $instance, $langcode, $items, $delta, $element) {
+  $icon = array(
+    '#type' => 'select',
+    '#options' => stanford_icon_grid_get_icons(),
+    '#attributes' => array('class' => array('stanford-fontawesome-icon')),
+    '#attached' => array(
+      'css' => array(
+        drupal_get_path('module', 'stanford_icon_grid') . '/css/fontawesome/stanford_fa.css',
+      ),
+      'js' => array(
+        drupal_get_path('module', 'stanford_icon_grid') . '/js/stanford_icon_grid.js',
+      ),
+    ),
+    '#default_value' => $items[$delta]['icon'],
+  );
+  return array('icon' => $element + $icon);
+}
+
+/**
+ * Gets available font awesome icons from the json file.
+ *
+ * @return array
+ *   Font awesome options in groups.
+ */
+function stanford_icon_grid_get_icons() {
+  if ($cache = cache_get('stanford_icon_grid_icons')) {
+    return $cache->data;
+  }
+
+  $fa_icons = drupal_get_path('module', 'stanford_icon_grid') . '/css/font-awesome.json';
+  $fa_icons = drupal_json_decode(file_get_contents($fa_icons));
+  cache_set('stanford_icon_grid_icons', $fa_icons);
+  return $fa_icons;
+}
+
+/**
+ * Implements hook_field_formatter_info().
+ */
+function stanford_icon_grid_field_formatter_info() {
+  return array(
+    'fontawesome_icon_formatter' => array(
+      'label' => t('Font Awesome Icons'),
+      'field types' => array('fontawesome_icon'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ */
+function stanford_icon_grid_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+  $element = array();
+  foreach ($items as $delta => $item) {
+    if (isset($item['icon'])) {
+      $element[$delta] = array(
+        '#theme' => 'html_tag',
+        '#tag' => 'i',
+        '#value' => '',
+        '#attributes' => array(
+          'class' => array('fa', 'fa-' . $item['icon']),
+          'aria-hidden' => array("true"),
+        ),
+      );
+      drupal_add_css(drupal_get_path('module', 'stanford_icon_grid') . '/css/fontawesome/stanford_fa.css');
```

Comment:
Does $element support $element['#attached']['css'][] = ... path to file? I would rather see an attached statement than a drupal_add_css.